src/graphics.rs

```diff
@@ -1628,6 +1634,263 @@ impl Drawable for Sprite {
     }
 }
 
+// AsciiSubset refers to the subset of ascii characters which give alphanumeric characters plus symbols
+pub enum CharacterListType {
+    AsciiSubset,
+    Ascii,
+    Cyrillic,
+    Thai,
+    Vietnamese,
+    Chinese,
+    Japanese,
+}
+
+#[derive(Debug, Clone)]
+struct CharInfo {
+    vertical_offset: i32,
+    width: f32,
+    uvs: Box2<f32>,
+}
+
+/// `FontTexture` is a texture generated using the *_character_list functions.
+/// It contains a texture representing all of the rasterized characters
+/// retrieved from the *_character_list function. `font_map` represents a
+/// a mapping between a character and its respective character texture
+/// located within `font_texture`.
+#[derive(Debug)]
+pub struct FontAtlas {
+    font_map: HashMap<char, CharInfo>,
+    font_texture: Cached<Texture>,
+}
+
+impl FontAtlas {
+    pub fn new<R: Read>(
+        ctx: &mut Graphics,
+        font: R,
+        font_size: f32,
+        char_list_type: CharacterListType,
+    ) -> Result<FontAtlas> {
+        let inval_bb = ::rusttype::Rect {
+            min: ::rusttype::Point { x: 0, y: 0 },
+            max: ::rusttype::Point {
```

Comment:
as of Rust edition 2018, `::` is no longer necessary when referring to extern crate symbols. `rusttype::Rect` and `rusttype::Point` is fine, and I would even say that there's good reason to put `use rusttype as rt;` at the start of this function so that you can shorten to `rt::Point` and `rt::Rect`. Also worth noting that rusttype has a `rusttype::point(x, y)` function explicitly for the purpose of making constructing points easier. Not sure if `Rect` has an equivalent.